### PR TITLE
Build ubuntu fips framework image

### PR DESCRIPTION
**Pull request by jimmykarily**
to allow users to build fips enabled ubuntu flavors from scratch

https://kairos.io/docs/reference/build-from-scratch/

Fixes https://github.com/kairos-io/kairos/issues/1539

### CODE REVIEW

File: .github/workflows/image.yaml [hunk 1]
```diff
@@ -38,6 +38,7 @@ jobs:
     runs-on: ubuntu-latest
     needs:
       - get-matrix
+    if: ${{ matrix.frameworkonly != "true" }}
```
Comment (Itxaka) [inline .github/workflows/image.yaml:41]: This wont work,  I dont think you can access the matrix at the job level :(

File: framework-profile.yaml [hunk 2]
```diff
@@ -84,6 +119,31 @@ opensuse-leap-kernel:
...
+  - fips/kcrypt-challenger
+  - fips/immucore
+  - fips/kairos-agent
+
+kairos-toolchain-nonfips:
+  packages:
+  - system/kcrypt
+  - system/kcrypt-challenger
+  - system/suc-upgrade
+  - system/grub2-efi
```
Comment (Itxaka) [inline framework-profile.yaml:144]: dup

File: framework-profile.yaml [hunk 3]
```diff
@@ -84,6 +119,31 @@ opensuse-leap-kernel:
...
+  - fips/kcrypt
+  - fips/kcrypt-challenger
+  - fips/immucore
+  - fips/kairos-agent
+
+kairos-toolchain-nonfips:
+  packages:
+  - system/kcrypt
+  - system/kcrypt-challenger
+  - system/suc-upgrade
```
Comment (Itxaka) [inline framework-profile.yaml:143]: dup

File: .github/workflows/image.yaml (hunk 1 above)
Comment (jimmykarily) [reply .github/workflows/image.yaml:41]: yes I see the failure now :( this is unfortunate

File: framework-profile.yaml (hunk 3 above)
Comment (jimmykarily) [reply framework-profile.yaml:143]: true, fixing.

File: framework-profile.yaml (hunk 2 above)
Comment (jimmykarily) [reply framework-profile.yaml:144]: fixing.

File: .github/workflows/image.yaml [hunk 4]
```diff
@@ -34,6 +35,27 @@ jobs:
...
+      run: |
+          content=`cat .github/flavors.json`
+          # the following lines are only required for multi line json
+          # the following lines are only required for multi line json
+          content="${content//'%'/'%25'}"
+          content="${content//$'\n'/'%0A'}"
+          content="${content//$'\r'/'%0D'}"
+          # end of optional handling for multi line json
+          # end of optional handling for multi line json
+          echo "::set-output name=matrix::{\"include\": $content }"
```
Comment (mauromorales) [inline .github/workflows/image.yaml:57]: nothing against this but just a reminder that these will be deprecated soon https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

Comment (jimmykarily) [reply .github/workflows/image.yaml:57]: Yes I know. I've had enough of Github Actions for today though :D Let's fix all instances of this in another PR.

Comment (mauromorales) [reply .github/workflows/image.yaml:57]: sounds good

File: framework-profile.yaml [hunk 5]
```diff
@@ -84,6 +119,29 @@ opensuse-leap-kernel:
...
+  - system/grub2-efi
+
+kairos-toolchain-fips:
+  packages:
+  - fips/kcrypt
+  - fips/kcrypt-challenger
+  - fips/immucore
+  - fips/kairos-agent
+
+kairos-toolchain-nonfips:
```
[the remark this answers is not in the document]
Comment (jimmykarily) [reply framework-profile.yaml:139]: true, changing

**Review by mauromorales — DISMISSED** [review]
lgtm

**Review by Itxaka — APPROVED** [review]

**Comment by jimmykarily** [comment]
I sent a commit to fix the broken test. Won't wait for CI to save some time and get the framework image build (only happens on `master`). If it breaks, I'll fix it on master.

**Pull request merged**